lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -181,6 +183,284 @@
+                        array[k++] = leftElement;
+                        if (j < bufferLen) {
+                            leftString = stringBuffer[j];
+                            leftElement = buffer[j++];
+                        } else {
+                            break;
+                        }
+                    }
+                }
+
+                while (j < bufferLen) {
+                    compareArray[k] = stringBuffer[j];
+                    array[k++] = buffer[j++];
+                }
+                --position;
+            } else {
+                const len = right - left;
+                current[3] = 1;
+                if (len > 2) {
+                    if (position < stack.length) {
+                        stack[position][0] = left;
+                        stack[position][1] = mid + left >> 1;
+                        stack[position][2] = mid;
+                        stack[position][3] = 0;
+                    } else {
+                        stack[position] = [left, mid + left >> 1, mid, 0];
+                    }
+                    ++position;
+                    if (fast)
+                    {
+                        if (position < stack.length) {
+                            stack[position][0] = mid;
+                            stack[position][1] = mid + right >> 1;
+                            stack[position][2] = right;
+                            stack[position][3] = 0;
+                        } else {
+                            stack[position] = [mid, mid + right >> 1, right, 0];
+                        }
+                    } else {
+                        if (position < stack.length) {
+                            stack[position][0] = mid;
+                            stack[position][1] = (mid + right) /2 |0;
+                            stack[position][2] = right;
+                            stack[position][3] = 0;
+                        } else {
+                            stack[position] = [mid, (mid + right) /2 |0, right, 0];
+                        }
+                    }
+                    ++position;
+                } else {
+                    if (len === 2) {
+                        if (compareArray[right - 1] < compareArray[left]) {
+                            const cache = array[left];
+                            array[left] = array[right - 1];
+                            array[right - 1] = cache;
+                            const  stringCache = compareArray[left];
+                            compareArray[left] = compareArray[right - 1];
+                            compareArray[right - 1] = stringCache;
+                        }
+                    }
+                    --position;
+                }
+            }
+        }
+    });
+
+    platform.registerChakraLibraryFunction("MergeSort", function(array, length, compareFn) {
+        const stack = __chakraLibrary.createObject(null);
+        const buffer = __chakraLibrary.createObject(null);
```

Comment:
I use Objects with null prototype for my stack and buffer to avoid any oddities from edited prototypes.